### DISCUSSION TITLE
Table get all filtered rows

### DISCUSSION THREAD

**Discussion by michaelcsikos** [ant-design-blazor/ant-design-blazor#3743] [Q&A] [answered]
How can I get all the rows on all pages when filters have been applied? `SelectAll()` only selects the rows in the current page. I tried temporarily setting the `PageSize` to `int.MaxValue` but it doesn't work.

I want to be able to filter a table with multiple columns and then export the data to a CSV.

**Comment by ElderJames** [comment]
You can get the QueryModel from table instance, then query all data form your datasource with the filterModels outside.

**Reply by michaelcsikos** [reply, marked as answer]
Hi @ElderJames , OK, I'm using the following:
```cs
var queryModel = _table.GetQueryModel() as QueryModel<Info>;

var list       = queryModel.ExecuteQuery(_table.DataSource.AsQueryable());
```

`Info` is the read-only object type bound to the `Table`.

However, if any column has been sorted it's throwing the following exception:
```cs
System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null. Arg_ParamName_Name
  Source=System.Private.CoreLib
  StackTrace:
   at System.ArgumentNullException.Throw(String paramName)
   at System.ArgumentNullException.ThrowIfNull(Object argument, String paramName)
   at System.Linq.Queryable.OrderBy[TSource,TKey](IQueryable`1 source, Expression`1 keySelector)
   at AntDesign.TableModels.SortModel`1.AntDesign.TableModels.ITableSortModel.SortList[TItem](IQueryable`1 source)
   at AntDesign.TableModels.QueryModel`1.ExecuteQuery(IQueryable`1 query)
```

**Reply by ElderJames** [reply]
Could you give me the complete code to reproduce the exception?

**Reply by michaelcsikos** [reply]
I think the sorting works for us now. We added it to our project with version 0.19.1. We were on 0.18.1 before.
